.github/workflows/ci.yml

```diff
@@ -0,0 +1,83 @@
+name: Build package on different Os and Python versions
+
+on : [push, pull_request]
+
+jobs:
+
+  build_package:
+    name: Build ${{ github.event.repository.name }} on ${{ matrix.os }} for Python-${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: [3.6, 3.7, 3.8, 3.9]
+        include:
+          - os: ubuntu-latest
+            path: ~/.cache/pip
+          - os: macos-latest
+            path: ~/Library/Caches/pip
+          - os: windows-latest
+            path: ~\AppData\Local\pip\Cache
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python-${{ matrix.python-version }} 
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Activating Python cache
+        uses: actions/cache@v2
+        id: cache_python
+        continue-on-error: true
+        with:
+          path: ${{ matrix.path }}
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('**/requirements.txt') }}
+
+      - name: Install dependencies for ${{ github.event.repository.name }}
+        run: |
+          # The --upgrade and --upgrade-strategy eager flags ensure that pip will always install the latest allowed version of all the dependencies.
+          python -m pip install --upgrade --upgrade-strategy eager -r requirements.txt
```

Comment:
The requirements are installed when installing the library, so isn't this step redundant?

Reply:
`--upgrade --upgrade-strategy eager` forces to download latest version of the packages if they are already installed in the machine.

But in this case, it's the "first" time they are install, for each time the CI is run. So I think that you're are right and this step could be removed.

Reply:
Done and it works ! :)